docs/openapi-docs/src/main/scala/tapir/docs/openapi/schema/ObjectSchemasForEndpoints.scala

```diff
@@ -37,35 +38,38 @@ object ObjectSchemasForEndpoints {
       .infoToKey
   }
 
-  private def objectSchemas(schema: TSchema): List[TSchema.SObject] = {
-    schema match {
-      case p: TSchema.SProduct =>
-        List(p) ++ p.fields
-          .map(_._2)
-          .flatMap(objectSchemas)
+  private def objectSchemas(schema: TSchema, validator: Validator[_]): List[(TSchema.SObject, Validator[_])] = {
+    (schema, validator) match {
+      case (p: TSchema.SProduct, v: ProductValidator[_]) =>
```

Comment:
I suppose we could also have cases where we have a `ValueValidator` for a product (e.g. custom validation)

Reply:
I suppose you are right, unfortunately ;)